platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/EspressoTest.java

```diff
@@ -1,24 +1,28 @@
 package com.mapbox.mapboxsdk.testapp.activity;
 
-import android.support.test.espresso.IdlingResource;
+import android.support.annotation.UiThread;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.Style;
 import com.mapbox.mapboxsdk.testapp.activity.espresso.EspressoTestActivity;
-import com.mapbox.mapboxsdk.testapp.utils.LoadStyleIdlingResource;
+
 
 /**
  * Base class for all tests using EspressoTestActivity as wrapper.
  * <p>
- * Uses {@link LoadStyleIdlingResource} to load "assets/streets.json" as style.
+ * Loads "assets/streets.json" as style.
  * </p>
  */
 public class EspressoTest extends BaseTest {
 
   @Override
-  protected IdlingResource generateIdlingResource() {
-    return new LoadStyleIdlingResource(rule.getActivity());
+  protected final Class getActivityClass() {
+    return EspressoTestActivity.class;
   }
 
+  @UiThread
   @Override
-  protected final Class getActivityClass() {
-    return EspressoTestActivity.class;
+  protected void initMap(MapboxMap mapboxMap) {
+    super.initMap(mapboxMap);
```

Comment:
I think this should be invoked _after_ setting the local style to ensure that we wait for it to load.

Reply:
the order doesn't matter, `getStyle` called as part `super.initMap` is async

Reply:
If the style that is a part of the hosting activity has already loaded `getStyle` will return immediately and count down the latch, without waiting for the below JSON style to load.

Reply:
that would be misusing EspressoTest imo, for those cases we should be using BaseTest instead.  Will change the execution order to support this behavior.

Reply:
:+1: thank you